Fix artifact generation to support multi TFM

kshyju opened the pull request:
Addresses part of #10044 

Fix artifact generation to support multi TFM.

Sample run: https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=165387&view=artifacts&pathAsName=false&type=publishedArtifacts

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

kshyju committed:
Multi TFM artifact creation

kshyju committed:
Minor udpate to the task display names

jviau commented:
I notice we are packing separate nuget artifacts for net6 and net8. This isn't really necessary as a nuget package can contain multiple TFMs with in (and the best match will be referenced during build). 

Do we want to ship separate packages, or are we intended to ship 1?

kshyju committed:
Removed duplicate nuget package artifact publish because we multi TFM

kshyju committed:
Do nuget push only for 8X packages.

kshyju commented:
> I notice we are packing separate nuget artifacts for net6 and net8. This isn't really necessary as a nuget package can contain multiple TFMs with in (and the best match will be referenced during build).
> 
> Do we want to ship separate packages, or are we intended to ship 1?

Yea, that makes sense. I removed the redundant NuGet packages & also updated the nuget push step.
